omc/core/model/board.py

```diff
@@ -0,0 +1,545 @@
+            raise NegativePiecePlayerNumException(
+                "Error: Player controller for piece"
+                f" {self.__class__.__name__} is negative. Player controller"
+                f" is {player_controller} which is less than 0.")
+
+        if player_controller not in self._board.current_players:
+            raise NonCurrentPiecePlayerNumException(
+                "Error: Player controller for piece"
+                f" {self.__class__.__name__} is non-current. Player"
+                f" controller is {player_controller} which is not a current"
+                f" player: {self._board.current_players}.")
+
+        self._player_controller = player_controller
+
+    @property
+    def piece_char(self) -> str:
+        """
+        Gets the character representation for the piece.
+
+        :return: 1-character string that represent the piece in the command
+            line display
+        :rtype: str
+        """
+        return self._piece_char
+
+    @property
+    def piece_pxl_hex(self) -> int:
+        """
+        Gets the pixel-hex representation for the piece.
+
+        :return: Hex value of 64-bit number representing an 8x8 pixel grid to
+            display the piece in a GUI.
+        :rtype: int
+        """
+
+        return self._piece_pxl_hex
+
+    @property
+    def current_players(self) -> list[int]:
+        """
+        Gets the list of current players from board.
+
+        :return: List of the current player numbers
+        :rtype: list[int]
+        """
+
+        return list(self._board.current_players.keys())
+
+    @property
+    def current_coords(self) -> ndarray[int]:
+        """
+        Gets the current coordinates of the piece
+
+        :return: Current player coordinate
+        :rtype: ndarray[int]
+        """
+        return self._current_coords
+
+    @property
+    def player_controller(self) -> int:
+        """
+        Gets the current player controller.
+
+        :return: Number associated with player currently controlling the piece.
+        :rtype: int
+        """
+        return self._player_controller
+
+    def list_moves(self) -> list[ndarray[int] | None]:
+        """
+        Gives a list of all available moves for the piece.
+
+        :return: List of moves currently able to be performed by
+            the piece.
+        :rtype: list[Optional[ndarray[int]]]
+        """
+
+        raise NotImplementedError
+
+    def move(self, coords: ndarray[int]) -> bool:
+        """
+        Performs an action on the piece using arguments.
+
+        :param coords: Coordinates for the space of the attempted move.
+        :type coords: ndarray[int]
+        :return: True on successful move, False otherwise.
+        :rtype: bool
+        """
+
+        # Check if move is valid
+        if coords not in self.list_moves():
+            print("Invalid Move.")
+            return False
+
+        self._current_coords = coords
+        return True
```

Comment:
This does not update the board's layout with the updated coordinates. Add something like
```
self._board.remove_piece(self)
self._current_coords = coords
self._board.add_piece(self)
```

Reply:
I don't think we should remove and re-add the piece, in case removal will have any future side-effects. We should make a `move_piece` function and use that